### PR TITLE
Victor kills nodes

### PR DESCRIPTION
**Pull request by vbustamante**
Remove Node and NodeId from the system completely

### CODE REVIEW

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

File: app/web/src/components/ModelingDiagram/diagram_types.ts [hunk 1]
```diff
@@ -98,45 +96,45 @@ export class DiagramEdgeData extends DiagramElementData {
...
 
   get toNodeKey() {
-    const comp = useComponentsStore().componentsByNodeId[this.def.toNodeId];
+    const comp = useComponentsStore().componentsById[this.def.toComponentId];
     if (comp?.isGroup) {
-      return DiagramGroupData.generateUniqueKey(this.def.toNodeId);
+      return DiagramGroupData.generateUniqueKey(this.def.toComponentId);
     }
-    return DiagramNodeData.generateUniqueKey(this.def.toNodeId);
+    return DiagramNodeData.generateUniqueKey(this.def.toComponentId);
```
Comment (britmyerss) [inline app/web/src/components/ModelingDiagram/diagram_types.ts:111]: is it worth replacing these unique keys with the component pks?

Comment (vbustamante) [reply app/web/src/components/ModelingDiagram/diagram_types.ts:111]: I don't think so. If we are in a changeset, any unchanged components will just refer to the entry on head, so they'll have a different PK, but the same ID. The uniqueness here is among different components and pk distinguishes amongst "versions"  of components too

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors try

**Comment by nickgerace** [comment]
@vbustamante we should consider putting this PR into drafts or closing it because we will want parts of it (mainly or all `app/web`) once we switchover to the new engine.

**Comment by vbustamante** [comment]
bors try

**Comment by vbustamante** [comment]
bors merge

**Comment by vbustamante** [comment]
bors merge

**Comment by vbustamante** [comment]
bors merge

**Pull request merged**